interfaces/cython/cantera/onedim.pyx

```diff
@@ -1003,7 +1028,13 @@ cdef class Sim1D:
 
         # Final call with expensive options enabled
         if have_user_tolerances or solve_multi or soret_doms:
-            self.sim.solve(loglevel, <cbool>refine_grid)
+            try:
+                self.sim.solve(loglevel, <cbool>refine_grid)
+            except Exception as e:
+                # restore settings before re-raising exception
+                set_transport(self.gas, self.domains, solve_multi)
+                set_soret(soret_doms, solve_soret)
+                raise e
```

Comment:
I'm not sure I understand the need for this `try/except` block -- the settings should have their intended final values at this point whether or not this final call to `solve` succeeds, right?

Reply:
correct.